src/extension.ts

```diff
@@ -430,11 +430,20 @@ export async function activate(context: vscode.ExtensionContext): Promise<void>
         return new LanguageClient('Phan Language Server', serverOptions, clientOptions);
     };
 
+    const languageClients = analyzedProjectDirectories.map(createClient);
+
+    async function restartLanguageClients(): Promise<void> {
+        for (const client of languageClients) {
```

Comment:
`await Promise.all(languageClients.map(client => client.restart()))` seems better, it would restart clients in parallel and restart faster

https://github.com/microsoft/vscode-languageserver-node/blob/292dd0d88b6465337e42ba40ad986fdfdeb42d4d/client/src/node/main.ts#L189-L201 as long as the languageserver dependency is new enough.

Reply:
Unfortunately, `restart` doesn't seem to be in the latest version of `vscode-languageserver-types`. Restarting in parallel seems reasonable, though.